utils/wasm-gen/src/wasm.rs

```diff
@@ -167,6 +167,14 @@ impl WasmModule {
     }
 }
 
+impl Deref for WasmModule {
```

Comment:
It's against design. Please remove the `Deref`.

Reply:
It provides readonly access to the underlying struct so why not to give it to the user? Anyway it already have method into_inner that can be used to expose underlying type

Reply:
calling `into_inner` is a signal that wasm generation work-flow is done. Check generatorts transitions.

Reply:
User must not do anything with `Module` when he gives it to generator.

Reply:
It seems to look legit because `Deref` provides immutable reference to `Module`. So, user will not be able to do anything with it.